src/renderer/components/DirectoryPicker/DirectoryPicker.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { Button } from '@mui/material';
+import TextField from '../TextField/TextField';
+
+interface Props {
+  value?: string;
+  placeholder?: string;
+  onClick?: () => void;
+}
+
+const DirectoryPicker: React.FC<Props> = ({ value, placeholder, onClick }) => {
+  const [currentValue, setCurrentValue] = React.useState(value);
+
+  const handleDirectoryChange = (event: React.ChangeEvent<HTMLInputElement>) =>
+    setCurrentValue(event.target.value);

```

Comment:
handleDirectoryChange in this case will just come as a prop, the same way you handled onClick. Thus, this piece of code becomes redundant